packages/map-template/src/components/ContactActionButton/ContactActionButton.jsx

```diff
@@ -0,0 +1,64 @@
+import PropTypes from "prop-types";
+import './ContactActionButton.scss'
+
+const detailTypes = {
+    email: 'email',
+    phone: 'phone',
+};
+
+ContactActionButton.propTypes = {
+    detailType: PropTypes.string,
+    active: PropTypes.bool,
+    displayText: PropTypes.string,
+    value: PropTypes.string,
+    icon: PropTypes.string
+};
+
+/**
+ * A button component that renders different types of contact actions (email, phone, website)
+ * with appropriate icons and click handlers. The button can open email clients, phone dialers,
+ * or external links based on the contact type.
+ * 
+ * @param {object} props
+ * @param {string} props.detailType - The type of the contact detail (e.g. email, phone, website).
+ * @param {boolean} props.active - Whether the button should be displayed.
+ * @param {string} props.displayText - The text to display on the button.
+ * @param {string} props.value - The value of the contact detail (e.g. email address, phone number, website URL).
+ * @param {string} props.icon - The icon to display on the button.
+ * @returns 
+ */
+function ContactActionButton({ detailType, active, displayText, value, icon }) {
+    if (!active) return null;  // Early exit to not render the button if it is not set as active in the additional details data
+    const textToDisplay = detailType.toLowerCase() === 'phone' ? value : displayText; // If the detail type is a phone number, display the phone number instead of the display text
+
+    const handleClick = (detailType, value) => {
```

Comment:
Could it be named generateURL or something similar since it no longer handles a click?